### DISCUSSION TITLE
lots of errors on install (New Macbook M1pro)

### DISCUSSION THREAD

**Discussion by Seriousaccident** [Homebrew/discussions#2712] [Getting started] [answered]
Hi, total rookie here so I have no idea what I'm doing. First time using a terminal and I tried installing Homebrew. It installed just fine, but I made a mistake with something unrelated and ended up uninstalling it for a stupid reason and now I'm trying to reinstall it and I'm getting errors.

When I uninstalled it, I followed someone's instruction to paste the install command into the terminal and just change the word install to uninstall. That seemed to work, but it said that a few files in the "opt" directory would not be removed and I could remove them manually if I wanted. I found those files and deleted them.

Now I am reinstalling and I get this: 
```
==> Downloading and installing Homebrew...
HEAD is now at 26bffd405 Merge pull request #12656 from Homebrew/dependabot/bundler/Library/Homebrew/rubocop-1.24.1
error: Not a valid ref: refs/remotes/origin/master
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
fatal: Could not resolve HEAD to a revision
==> Installation successful!
```

Since it said "Installation successful" I continued to follow the instructions, but when I tried to install node, it said this:

```
Running `brew update --preinstall`...
fatal: Could not resolve HEAD to a revision
==> Tapping homebrew/cask
Cloning into '/opt/homebrew/Library/Taps/homebrew/homebrew-cask'...
remote: Enumerating objects: 612948, done.
remote: Total 612948 (delta 0), reused 0 (delta 0), pack-reused 612948
Receiving objects: 100% (612948/612948), 276.45 MiB | 1.12 MiB/s, done.
Resolving deltas: 100% (433476/433476), done.
Tapped 3956 casks (4,029 files, 296MB).
Warning: No available formula with the name "node".
==> Searching for similarly named formulae...
Error: No similarly named formulae found.
It was migrated from homebrew/cask to homebrew/core.
```

I'm totally lost obviously, so can anyone help?
Thank you

**Comment by gromgit** [comment, marked as answer]
What's the output of `brew config` and `brew doctor`?

**Reply by Seriousaccident** [reply]
Thanks for the reply. Brew config:

`HOMEBREW_VERSION: 3.3.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 96137bc19e68398ebbb7033379df288cd8b9a3f9
Last commit: 2 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: HEAD
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.32.0 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.0.1-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: N/A
Rosetta 2: false
`

Brew doctor:

`Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew/homebrew-core was not tapped properly! Run:
  rm -rf "/opt/homebrew/Library/Taps/homebrew/homebrew-core"
  brew tap homebrew/core

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout master`

**Reply by carlocab** [reply]
Have you tried following this suggestion from `brew doctor`?
```
rm -rf "/opt/homebrew/Library/Taps/homebrew/homebrew-core"
brew tap homebrew/core
```

**Reply by Seriousaccident** [reply]
ok, I did that and it did this:

`==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
  https://github.com/Homebrew/brew#donations
Running `brew update --preinstall`...
==> Tapping homebrew/core
Cloning into '/opt/homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 1121165, done.
remote: Counting objects: 100% (10/10), done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 1121165 (delta 4), reused 4 (delta 0), pack-reused 1121155
Receiving objects: 100% (1121165/1121165), 436.43 MiB | 960.00 KiB/s, done.
Resolving deltas: 100% (774245/774245), done.
Tapped 3 commands and 5911 formulae (6,241 files, 480.3MB).
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/cask).
==> New Casks
direqual                                                                     openin                                                                       restream-chat
==> Updated Casks
Updated 36 casks.
`

Do you think that fixed things? Should I try install node now?

**Reply by carlocab** [reply]
It seems so. What's the output of `brew config` and `brew doctor` now? Wrap the output in three backticks (```) on the line before and the line after to render the output properly. You're currently using only one backtick, and that makes things hard to read.

**Reply by Seriousaccident** [reply]
```HOMEBREW_VERSION: 3.3.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 96137bc19e68398ebbb7033379df288cd8b9a3f9
Last commit: 2 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9319d2095aec692c9d130e7d71f28de6fd6e6822
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.32.0 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.0.1-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: N/A
Rosetta 2: false
```

And

```Your system is ready to brew.```

**Reply by carlocab** [reply]
You're all set.

**Reply by Seriousaccident** [reply]
Thank you!

**Reply by Seriousaccident** [reply]
Hang on. I installed node and then tried to install iterm2:

``` 
brew cask install iterm2
Error: Unknown command: cask

**Reply by gromgit** [reply]
The `brew cask` subcommand was deprecated a long time ago. Try:
```
brew install --cask iterm2
```

**Reply by Seriousaccident** [reply]
Ah, ok. Thanks.

**Reply by carlocab** [reply]
`brew install iterm2` also works.
